components/support/jwcrypto/src/ec.rs

```diff
@@ -23,12 +22,15 @@ pub struct ECKeysParameters {
 
 pub(crate) fn encrypt_to_jwe(
     data: &[u8],
-    encryption_params: EncryptionParameters,
+    enc: EncryptionAlgorithm,
+    peer_jwk: &Jwk,
 ) -> Result<CompactJwe> {
-    let EncryptionParameters::ECDH_ES { enc, peer_jwk } = encryption_params;
     let local_key_pair = EphemeralKeyPair::generate(&agreement::ECDH_P256)?;
     let local_public_key = extract_pub_key_jwk(&local_key_pair)?;
-    let JwkKeyParameters::EC(ref ec_key_params) = peer_jwk.key_parameters;
+    let ec_key_params = match peer_jwk.key_parameters {
+        JwkKeyParameters::EC(ref params) => params,
+        JwkKeyParameters::Direct(_) => return Err(JwCryptoError::IllegalState("Not an EC key")),
```

Comment:
nit: if you use a default `_` case (or maybe an `if let`?) here you won't have to update this again later when we add more key types.